unepwcmc/web-components
# feat: add admin authentication to CUD table actions

Designed to be compatible with `wcmc_devise_sso` which adds role `wcmc` to users logging in via SSO: https://github.com/unepwcmc/web-components/pull/95

I was tempted to add in a line allowing authentication is `user.admin?` is true, but thought better to be more secure in case this is used on sites that have multiple admin layers, one of which shouldn't be able to access routes. Easier to add a user role to that site that overwrite the gem's authentication method.

Codebase: https://unep-wcmc.codebasehq.com/projects/editable-loadable-filterable/tickets/42

Would be good to test with the SSO branch when that is reviewed and working. #95 

To test
- Try and access new/edit/create routes when not logged in. You should be redirected to homepage (or sign in if devise is set up)
- Set some records to archived in the db e.g. Country.first.update(archived: true) and check these aren't pulled through in the api or table index action
- Hard to test with a logged in user until the SSO work is merged and that can be used. For now you can force the authentication method to return true as a half-test
